lib/embulk/input/mixpanel.rb

```diff
@@ -29,17 +29,32 @@ class Mixpanel < InputPlugin
       # between each 7 (SLICE_DAYS_COUNT) days.
       SLICE_DAYS_COUNT = 7
 
+      DEFAULT_TIME_COLUMN = 'time'
+
       def self.transaction(config, &control)
         timezone = config.param(:timezone, :string)
         TimezoneValidator.new(timezone).validate
 
         from_date = config.param(:from_date, :string, default: (Date.today - 2).to_s)
         fetch_days = config.param(:fetch_days, :integer, default: nil)
-        range = RangeGenerator.new(from_date, fetch_days).generate_range
-        Embulk.logger.info "Try to fetch data from #{range.first} to #{range.last}"
+
 
         fetch_unknown_columns = config.param(:fetch_unknown_columns, :bool, default: false)
 
+        incremental_column = config.param(:incremental_column, :string, default: nil)
+        incremental = config.param(:incremental, :bool, default: true)
+        latest_fetched_time = config.param(:latest_fetched_time, :integer, default: 0)
+
+        # Backfill from date if incremental and an incremental field is set and we are in incremental run
+        if incremental && !incremental_column.nil? && latest_fetched_time !=0
```

Comment:
minor: `if !incremental_column.nil?` can be simplified as `if incremental_column`.